internal/eventstore/handler/handler_projection.go

```diff
@@ -209,32 +239,82 @@ func (h *ProjectionHandler) schedule(ctx context.Context) {
 				max = len(ids)
 			}
 			instances := ids[i:max]
-			lockCtx, cancelLock := context.WithCancel(ctx)
-			errs := h.lock(lockCtx, h.requeueAfter, instances...)
+			lockInstanceCtx, cancelInstanceLock := context.WithCancel(lockCtx)
+			errs := h.lock(lockInstanceCtx, h.requeueAfter, instances...)
 			//wait until projection is locked
 			if err, ok := <-errs; err != nil || !ok {
-				cancelLock()
+				cancelInstanceLock()
 				logging.WithFields("projection", h.ProjectionName).OnError(err).Warn("initial lock failed")
 				failed = true
 				continue
 			}
-			go h.cancelOnErr(lockCtx, errs, cancelLock)
-			err = h.Trigger(lockCtx, instances...)
+			go h.cancelOnErr(lockInstanceCtx, errs, cancelInstanceLock)
+			err = h.Trigger(lockInstanceCtx, instances...)
 			if err != nil {
 				logging.WithFields("projection", h.ProjectionName, "instanceIDs", instances).WithError(err).Error("trigger failed")
 				failed = true
 			}
 
-			cancelLock()
+			cancelInstanceLock()
 			unlockErr := h.unlock(instances...)
 			logging.WithFields("projection", h.ProjectionName).OnError(unlockErr).Warn("unable to unlock")
 		}
+		// if the first schedule did not fail, store that in the eventstore, so we can check on later starts
+		if !succeededOnce {
+			if !failed {
+				err = h.setSucceededOnce(ctx)
+				logging.WithFields("projection", h.ProjectionName).OnError(err).Warn("unable to push first schedule succeeded")
+			}
+			cancelLock()
+			unlockErr := h.unlock("system")
+			logging.WithFields("projection", h.ProjectionName).OnError(unlockErr).Warn("unable to unlock first schedule")
+		}
 		// it succeeded at least once if it has succeeded before or if it has succeeded now - not failed ;-)
 		succeededOnce = succeededOnce || !failed
 		h.triggerProjection.Reset(h.requeueAfter)
 	}
 }
 
+func (h *ProjectionHandler) hasSucceededOnce(ctx context.Context) (bool, error) {
+	events, err := h.Eventstore.Filter(ctx, eventstore.NewSearchQueryBuilder(eventstore.ColumnsEvent).
+		AddQuery().
+		AggregateTypes(aggregateType).
+		AggregateIDs(aggregateID).
+		EventTypes(schedulerSucceeded).
+		EventData(map[string]interface{}{
+			"name": h.ProjectionName,
+		}).
+		Builder(),
+	)
+	if err != nil {
```

Comment:
Simplify return by len(...) > 0 && err != nil, err

Reply:
And lines below

Reply:
`...&& err == nil...` is even more helpful 😉